lib/mix/tasks/phx.gen.notifier.ex

```diff
@@ -49,12 +54,20 @@ defmodule Mix.Tasks.Phx.Gen.Notifier do
 
     prompt_for_conflicts(context)
 
-    copy_new_files(context, binding, paths)
+    if Keyword.get(test_opts, :compile_app?, true) do
+      # This is needed in order to detect if the mailer is defined in the project.
+      Mix.Task.run("compile")
+    end
```

Comment:
Instead of test_opts, do this:

```suggestion
    if "--no-compile" in args do
      Mix.Task.run("compile")
    end
```